prov/efa/src/rxr/rxr_cq.c

```diff
@@ -213,16 +213,17 @@ void rxr_cq_write_tx_error(struct rxr_ep *ep, struct rxr_tx_entry *tx_entry,
 		break;
 	case RXR_TX_QUEUED_CTRL:
 	case RXR_TX_QUEUED_SHM_RMA:
-	case RXR_TX_QUEUED_REQ_RNR:
-	case RXR_TX_QUEUED_DATA_RNR:
-		dlist_remove(&tx_entry->queued_entry);
+		dlist_remove(&tx_entry->queued_ctrl_entry);
```

Comment:
I don't fully understand this section, what state are the RNR packets in this switch case?

Reply:
With this PR, RNR is not tracked by state, but tracked by the newly introduced flag `RXR_TX_ENTRY_QUEUED_RNR`.

Reply:
See line 224, where

```
if (tx_entry->rxr_flags & RXR_TX_ENTRY_QUEUED_RNR)
		dlist_remove(&tx_entry->queued_rnr_entry);
```

Reply:
If that's the case, wouldn't we fall into the default case and hit an assert?

Reply:
I am not sure I understand. Where will we hit the assert?

Reply:
William is asking if the tx_entry is for a RNR packet, since it's not tracked by state, will it fall into the default case (L218) and hit assert(0 && "tx_entry unknown state") (L221)

Reply:
I see. 
It will be still in one of the states RXR_TX_SEND or RXR_TX_REQ
An tx_entry in send or req state can both encounter RNR, before this patch their state will be changed to UEUE_DATA_RNR and QUEUE_CTRL_RNR. After this patch, when encounter RNR, their state remain unchanged, the flag RXR_TX_ENTRY_ENCOUTER_RNR is toggled.

So, in this code block, the state will be either RXR_TX_SEND or RXR_TX_REQ